### PR TITLE
Created more precise tests for the creation of sync locations

### PR DESCRIPTION
**Pull request by etorreborre**
I developed all the possible combinations of `location x sync-dir`.

### CODE REVIEW

File: notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala [hunk 1]
```diff
@@ -107,13 +111,47 @@ class SynchronizedInputsOutputsSpec extends AwsScalaCheckSpec(tests = 5) with Di
...
+      val syncLocation = createSynchronizedLocation(None, location).toOption.get
+      syncLocation.path ==== new Path(location.path)
+    } &&
+    // S3
+    prop { (location: S3Location, syncDir: ExecutionLocation) =>
+      val syncLocation = createSynchronizedLocation(Some(syncDir), location).toOption.get
+      val syncDirIsHdfs = syncDir.fold(_ => true, _ => false)
+
+      (syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path)).when(!syncDirIsHdfs) and
+      (syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path)).when(syncDirIsHdfs)
```
Comment (raronson) [inline notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala]: Could this be reduced to just `(syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path))`?

File: notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala [hunk 2]
```diff
@@ -107,13 +111,47 @@ class SynchronizedInputsOutputsSpec extends AwsScalaCheckSpec(tests = 5) with Di
...
     valid1 ==> valid2 :| "v1 ==> v2" &&
-    valid2 ==> valid1 :| "v2 ==> v1" &&
-    (!isHdfsNoSync ||
-      ((location, syncLocation.toOption) must beLike {
-        case (HdfsLocation(p), Some(sl)) => sl.path.toUri.toString ==== p
-      }))
+    valid2 ==> valid1 :| "v2 ==> v1"
   }
 
+  def syncPaths =
```
Comment (raronson) [inline notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala]: Can this be split into different tests?

Comment (etorreborre) [inline notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala]: Done.

File: notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala (hunk 1 above)
Comment (etorreborre) [inline notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala]: Done.

**Pull request merged**